src/simplewallet/simplewallet.cpp

```diff
@@ -1472,7 +1472,7 @@ bool simple_wallet::init(const boost::program_options::variables_map& vm)
     if (m_restoring)
     {
       uint64_t approximate_blockchain_height = simple_wallet::get_approximate_blockchain_height();
-      if (m_restore_height > approximate_blockchain_height){
+      if (m_restore_height >= approximate_blockchain_height){
```

Comment:
Now, we could introduce an extra arbitrary amount like `restore_height + arbitrary_sanity_value >= current_height` . Not sure if this would improve the sanity check.